<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Optimize the De Novo Stacks Pipeline via R • RADstackshelpR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Optimize the De Novo Stacks Pipeline via R">
<meta property="og:description" content='Offers a handful of useful wrapper functions which streamline
    the reading, analyzing, and visualizing of variant call format (vcf) files in R.
    This package was designed to facilitate an explicit pipeline for optimizing
    Stacks (Rochette et al., 2019) (&lt;doi:10.1111/mec.15253&gt;)
    parameters during de novo (without a reference genome) assembly and variant
    calling of restriction-enzyme associated DNA sequence (RADseq) data. The
    pipeline implemented here is based on the 2017 paper "Lost in Parameter Space"
    (Paris et al., 2017) (&lt;doi:10.1111/2041-210X.12775&gt;) which
    establishes clear recommendations for optimizing the parameters
    m, M, and n, during the process of assembling loci.'>
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">RADstackshelpR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="radstackshelpr-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#radstackshelpr-" class="anchor"></a>RADstackshelpR <img src="reference/figures/logo.png" align="right" alt="" width="120">
</h1></div>
<p>RADstackshelpR offers a handful of useful wrapper functions which streamline the reading, analyzing, and visualizing of variant call format (vcf) files in R. The internal calls of each function rely heavily on the excellent R package <a href="https://knausb.github.io/vcfR_documentation/">vcfR</a> to read in and analyze vcf files, and the widely renowned <a href="https://ggplot2.tidyverse.org/">ggplot2</a> package to create elegant visualizations. This package was designed to facilitate an explicit pipeline for optimizing <a href="https://catchenlab.life.illinois.edu/stacks/">STACKS</a> paramaters during de novo (without a reference genome) assembly and variant calling of restriction-enzyme associated DNA sequence (RADseq) data. STACKS is a relatively user-friendly, command-line program designed for assembling RADseq data into loci, and performing variant (SNP) calling. STACKS offers users flexibility in setting parameters during the assembly process, allowing custom parameter optimization for any input dataset. The pipeline implemented here is based on the 2017 paper <a href="https://doi.org/10.1111/2041-210X.12775">Lost in Parameter Space</a> which establishes clear recommendations for optimizing the parameters ‘m’, ‘M’, and ‘n’, during the process of assembling loci.</p>
<p>Despite these clear recommendations, the full range of parameter space suggested to explore in this paper is often left unexplored in empirical studies, due to the computational and logistical difficulty of executing and analyzing 16 separate runs through the entire STACKS pipeline. This package is designed to automate that logistical difficulty, leaving users with a clear set of steps to follow for thoroughly optimized, de novo RAD locus assembly and variant calling. Simply follow the steps below to run the optimized de novo assembly pipeline.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Install development version from GitHub</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"DevonDeRaad/RADstackshelpR"</span><span class="op">)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>The first step is demultiplexing your sequence data using the ‘process_radtags’ function from STACKS, which could be executed by running something like this in a terminal window, if your raw sequence file is in your working directory, and you used the enzyme ‘ndeI’ as your cutter:</p>
<div id="demultiplex" class="section level3">
<h3 class="hasAnchor">
<a href="#demultiplex" class="anchor"></a>Demultiplex</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1"></a><span class="ex">/home/path/to/stacks-2.41/process_radtags</span> -p .  -o . -b plate.1.barcodes.txt -e ndeI -r -c -q</span></code></pre></div>
<p>More details on demultiplexing using process_radtags can be found <a href="https://catchenlab.life.illinois.edu/stacks/comp/process_radtags.php">here</a></p>
</div>
<div id="iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module" class="section level3">
<h3 class="hasAnchor">
<a href="#iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module" class="anchor"></a>Iterate over potential values for the ‘m’ parameter in the ‘ustacks’ module</h3>
<p>Once you have an individual zipped fastq file for each sample, we need to iterate over the relevant values for ‘m’ within the ‘ustacks’ module (here using 15 threads at each step to speed up computation). Running the following code in a terminal window will perform five separate iterations of the entire STACKS pipeline, each with a different parameter setting for ‘m’ (3-7), and save the results as an unfiltered vcf file in a specified directory.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1"></a><span class="co">#Running in bash/shell</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co">#designate all sample ID's to a single variable called 'files', each sample should be in the directory, and the filename should match this designation except for the extension, e.g., 'sample_2' = 'sample_2.fq.gz'</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="va">files=</span><span class="st">"sample_1</span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="st">sample_2</span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="st">sample_3</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="st">sample_4</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="st">sample_5"</span></span>
<span id="cb3-8"><a href="#cb3-8"></a></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="co"># Build loci de novo in each sample for the single-end reads only.</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co"># -M — Maximum distance (in nucleotides) allowed between stacks (default 2).</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co"># -m — Minimum depth of coverage required to create a stack (default 3).</span></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="co">#here, we will vary m from 3-7, and leave all other paramaters default</span></span>
<span id="cb3-13"><a href="#cb3-13"></a></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{3..7}</span></span>
<span id="cb3-15"><a href="#cb3-15"></a><span class="kw">do</span></span>
<span id="cb3-16"><a href="#cb3-16"></a><span class="co">#create a directory to hold this unique iteration:</span></span>
<span id="cb3-17"><a href="#cb3-17"></a><span class="fu">mkdir</span> stacks_m<span class="va">$i</span></span>
<span id="cb3-18"><a href="#cb3-18"></a><span class="co">#run ustacks with m equal to the current iteration (3-7) for each sample</span></span>
<span id="cb3-19"><a href="#cb3-19"></a><span class="va">id=</span>1</span>
<span id="cb3-20"><a href="#cb3-20"></a><span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span></span>
<span id="cb3-21"><a href="#cb3-21"></a><span class="kw">do</span></span>
<span id="cb3-22"><a href="#cb3-22"></a>    <span class="ex">/home/path/to/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_m<span class="va">$i</span> -i <span class="va">$id</span> -m <span class="va">$i</span> -p 15</span>
<span id="cb3-23"><a href="#cb3-23"></a>    <span class="bu">let</span> <span class="st">"id+=1"</span></span>
<span id="cb3-24"><a href="#cb3-24"></a><span class="kw">done</span></span>
<span id="cb3-25"><a href="#cb3-25"></a><span class="co">## Run cstacks to compile stacks between samples. Popmap is a file in working directory called 'pipeline_popmap.txt'</span></span>
<span id="cb3-26"><a href="#cb3-26"></a><span class="ex">/home/path/to/stacks-2.41/cstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -p 15</span>
<span id="cb3-27"><a href="#cb3-27"></a><span class="co">## Run sstacks. Match all samples supplied in the population map against the catalog.</span></span>
<span id="cb3-28"><a href="#cb3-28"></a><span class="ex">/home/path/to/stacks-2.41/sstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -p 15</span>
<span id="cb3-29"><a href="#cb3-29"></a><span class="co">## Run tsv2bam to transpose the data so it is stored by locus, instead of by sample.</span></span>
<span id="cb3-30"><a href="#cb3-30"></a><span class="ex">/home/path/to/stacks-2.41/tsv2bam</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -t 15</span>
<span id="cb3-31"><a href="#cb3-31"></a><span class="co">## Run gstacks: build a paired-end contig from the metapopulation data (if paired-reads provided),</span></span>
<span id="cb3-32"><a href="#cb3-32"></a><span class="co">## align reads per sample, call variant sites in the population, genotypes in each individual.</span></span>
<span id="cb3-33"><a href="#cb3-33"></a><span class="ex">/home/path/to/stacks-2.41/gstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -t 15</span>
<span id="cb3-34"><a href="#cb3-34"></a><span class="co">## Run populations completely unfiltered and output unfiltered vcf, for input to the RADstackshelpR package</span></span>
<span id="cb3-35"><a href="#cb3-35"></a><span class="ex">/home/path/to/stacks-2.41/populations</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15</span>
<span id="cb3-36"><a href="#cb3-36"></a><span class="kw">done</span></span></code></pre></div>
<p>You should now have five directories, named: stacks_m3, stacks_m4, stacks_m5, stacks_m6, &amp; stacks_m7, each of which contains an vcf file with all called SNPs for the given parameter settings (i.e., stacks_m3 = the directory containing output from the iteration where ‘m’ was set to 3). Now we will use RADstackshelpR to determine which of these parameter settings (m = 3-7) is optimal for this dataset according to the ‘R80’ cutoff (see <a href="%3Chttps://doi.org/10.1111/2041-210X.12775">Lost in Parameter Space</a>). I have now moved each vcf file into a local directory, and named it according to the parameter settings for the given run.</p>
</div>
<div id="use-radstackshelpr-to-visualize-the-output-of-these-5-runs-and-determine-the-optimal-value-for-the-parameter-m" class="section level3">
<h3 class="hasAnchor">
<a href="#use-radstackshelpr-to-visualize-the-output-of-these-5-runs-and-determine-the-optimal-value-for-the-parameter-m" class="anchor"></a>Use RADstackshelpR to visualize the output of these 5 runs and determine the optimal value for the parameter ‘m’.</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#running in R</span>
<span class="co">#load RADstackshelpR package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RADstackshelpR</span><span class="op">)</span>

<span class="co">#optimize_m function will generate summary stats on your 5 iterative runs</span>
<span class="co">#input can be full path to each file, or just the file name if the vcf files are in your working directory</span>
<span class="va">m.out</span><span class="op">&lt;-</span><span class="fu"><a href="reference/optimize_m.html">optimize_m</a></span><span class="op">(</span>m3<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_3.vcf"</span>,
           m4<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_4.vcf"</span>,
           m5<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_5.vcf"</span>,
           m6<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_6.vcf"</span>,
           m7<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_7.vcf"</span><span class="op">)</span>

<span class="co">#Assigning the output of this function to the variable 'm.out' should generate a list containing five objects of class 'data.frame' with the following characteristics: 'depth' showing depth per sample for each m value, 'snp' showing the number of non-missing SNPs retained in each sample at each m value, 'loci' showing the number of non-missing loci retained in each sample at each m value, 'snp.R80' showing the total number of SNPs retained at an 80% completeness cutoff, and 'loci.R80' showing the total number of polymorphic loci retained at an 80% completeness cutoff.</span>

<span class="co">#Use this output list as input for this function, to visualize the effect of varying m on the depth of each sample</span>
<span class="fu"><a href="reference/vis_depth.html">vis_depth</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span><span class="op">)</span>
<span class="co">#&gt; [1] "Visualize how different values of m affect average depth in each sample"</span>
<span class="co">#&gt; Picking joint bandwidth of 9.53</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-5-1.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying m on the number of SNPs retained</span>
<span class="fu"><a href="reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 7190</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-5-2.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying m on the number of loci retained</span>
<span class="fu"><a href="reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 3420</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-5-3.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#3 is the optimal m value, and will be used next to optimize M</span></code></pre></div>
</div>
<div id="iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module-1" class="section level3">
<h3 class="hasAnchor">
<a href="#iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module-1" class="anchor"></a>Iterate over potential values for the ‘M’ parameter in the ‘ustacks’ module</h3>
<p>Now that we know the optimal value for ‘m’ is 3, we will repeat the process of iterating over parameter values in STACKS, this time varying the ‘M’ parameter from 1-8 within the ‘ustacks’ module. In this example, we again use 15 threads to speed up each step. Execute the following code in a terminal window:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1"></a><span class="co">#Running in bash/shell</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="co"># -M — Maximum distance (in nucleotides) allowed between stacks (default 2).</span></span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="co"># -m — Minimum depth of coverage required to create a stack (default 3).</span></span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co">#here, vary M from 1-8, and set m to the optimized value based on prior visualizations (here 3)</span></span>
<span id="cb8-5"><a href="#cb8-5"></a></span>
<span id="cb8-6"><a href="#cb8-6"></a><span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{1..8}</span></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="kw">do</span></span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="co">#create a directory to hold this unique iteration:</span></span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="fu">mkdir</span> stacks_bigM<span class="va">$i</span></span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="co">#run ustacks with M equal to the current iteration (1-8) for each sample, and m set to the optimized value (here, m=3)</span></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="va">id=</span>1</span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span></span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="kw">do</span></span>
<span id="cb8-14"><a href="#cb8-14"></a>    <span class="ex">/home/d669d153/work/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_bigM<span class="va">$i</span> -i <span class="va">$id</span> -m 3 -M <span class="va">$i</span> -p 15</span>
<span id="cb8-15"><a href="#cb8-15"></a>    <span class="bu">let</span> <span class="st">"id+=1"</span></span>
<span id="cb8-16"><a href="#cb8-16"></a><span class="kw">done</span></span>
<span id="cb8-17"><a href="#cb8-17"></a><span class="ex">/home/path/to/stacks-2.41/cstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -p 15</span>
<span id="cb8-18"><a href="#cb8-18"></a><span class="ex">/home/path/to/stacks-2.41/sstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -p 15</span>
<span id="cb8-19"><a href="#cb8-19"></a><span class="ex">/home/path/to/stacks-2.41/tsv2bam</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -t 15</span>
<span id="cb8-20"><a href="#cb8-20"></a><span class="ex">/home/path/to/stacks-2.41/gstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -t 15</span>
<span id="cb8-21"><a href="#cb8-21"></a><span class="ex">/home/path/to/stacks-2.41/populations</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15</span>
<span id="cb8-22"><a href="#cb8-22"></a><span class="kw">done</span></span></code></pre></div>
<p>You should now have eight directories, named: stacks_bigM1, stacks_bigM2, … stacks_bigM8,, each of which contains an vcf file with all called SNPs for the given parameter settings. I have again moved each vcf file into a local directory, and named it according to the parameter settings for the given run.</p>
</div>
<div id="use-radstackshelpr-to-visualize-the-output-of-these-8-runs-and-determine-the-optimal-value-for-the-parameter-m" class="section level3">
<h3 class="hasAnchor">
<a href="#use-radstackshelpr-to-visualize-the-output-of-these-8-runs-and-determine-the-optimal-value-for-the-parameter-m" class="anchor"></a>Use RADstackshelpR to visualize the output of these 8 runs and determine the optimal value for the parameter ‘M’.</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#running in R</span>
<span class="co">#optimize_bigM function will generate summary stats on your 8 iterative runs</span>
<span class="va">M.out</span><span class="op">&lt;-</span><span class="fu"><a href="reference/optimize_bigM.html">optimize_bigM</a></span><span class="op">(</span>M1<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M1.vcf"</span>,
           M2<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M2.vcf"</span>,
           M3<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M3.vcf"</span>,
           M4<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M4.vcf"</span>,
           M5<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M5.vcf"</span>,
           M6<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M6.vcf"</span>,
           M7<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M7.vcf"</span>,
           M8<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M8.vcf"</span><span class="op">)</span>

<span class="co">#Assigning the output of this function to the variable 'M.out' should generate a list containing four objects of class 'data.frame' with the following characteristics: 'snp' showing the number of non-missing SNPs retained in each sample at each m value, 'loci' showing the number of non-missing loci retained in each sample at each m value, 'snp.R80' showing the total number of SNPs retained at an 80% completeness cutoff, and 'loci.R80' showing the total number of polymorphic loci retained at an 80% completeness cutoff.</span>

<span class="co">#use this function to visualize the effect of varying 'M' on the number of SNPs retained</span>
<span class="fu"><a href="reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 6090</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-7-1.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying 'M' on the number of polymorphic loci retained</span>
<span class="fu"><a href="reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 2920</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-7-2.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">#optimal value for this dataset is M = 2</span></code></pre></div>
</div>
<div id="iterate-over-potential-values-for-the-n-parameter-in-the-cstacks-module" class="section level3">
<h3 class="hasAnchor">
<a href="#iterate-over-potential-values-for-the-n-parameter-in-the-cstacks-module" class="anchor"></a>Iterate over potential values for the ‘n’ parameter in the ‘cstacks’ module</h3>
<p>For the last optimization iteration, we will vary the ‘n’ parameter within the ‘cstacks’ module, using the previously optimized values for ‘m’ and ‘M’, by running the following code in a terminal window:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1"></a><span class="co">#Running in bash/shell</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="co"># -n — Number of mismatches allowed between sample loci when build the catalog (default 1).</span></span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="co">#here, vary 'n' across M-1, M, and M+1 (because my optimized 'M' value = 2, I will iterate over 1, 2, and 3 here), with 'm' and 'M' set to the optimized value based on prior visualizations (here 'm' = 3, and 'M'=2).</span></span>
<span id="cb12-4"><a href="#cb12-4"></a></span>
<span id="cb12-5"><a href="#cb12-5"></a><span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{1..3}</span></span>
<span id="cb12-6"><a href="#cb12-6"></a><span class="kw">do</span></span>
<span id="cb12-7"><a href="#cb12-7"></a><span class="co">#create a directory to hold this unique iteration:</span></span>
<span id="cb12-8"><a href="#cb12-8"></a><span class="fu">mkdir</span> stacks_n<span class="va">$i</span></span>
<span id="cb12-9"><a href="#cb12-9"></a><span class="co">#run ustacks with n equal to the current iteration (1-3) for each sample, m = 3, and M=2</span></span>
<span id="cb12-10"><a href="#cb12-10"></a><span class="va">id=</span>1</span>
<span id="cb12-11"><a href="#cb12-11"></a><span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span></span>
<span id="cb12-12"><a href="#cb12-12"></a><span class="kw">do</span></span>
<span id="cb12-13"><a href="#cb12-13"></a>    <span class="ex">/home/d669d153/work/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_n<span class="va">$i</span> -i <span class="va">$id</span> -m 3 -M 2 <span class="va">$i</span> -p 15</span>
<span id="cb12-14"><a href="#cb12-14"></a>    <span class="bu">let</span> <span class="st">"id+=1"</span></span>
<span id="cb12-15"><a href="#cb12-15"></a><span class="kw">done</span></span>
<span id="cb12-16"><a href="#cb12-16"></a><span class="ex">/home/path/to/stacks-2.41/cstacks</span> -n <span class="va">$i</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -p 15</span>
<span id="cb12-17"><a href="#cb12-17"></a><span class="ex">/home/path/to/stacks-2.41/sstacks</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -p 15</span>
<span id="cb12-18"><a href="#cb12-18"></a><span class="ex">/home/path/to/stacks-2.41/tsv2bam</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -t 15</span>
<span id="cb12-19"><a href="#cb12-19"></a><span class="ex">/home/path/to/stacks-2.41/gstacks</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -t 15</span>
<span id="cb12-20"><a href="#cb12-20"></a><span class="ex">/home/path/to/stacks-2.41/populations</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15</span>
<span id="cb12-21"><a href="#cb12-21"></a><span class="kw">done</span></span></code></pre></div>
</div>
<div id="use-radstackshelpr-to-visualize-the-output-of-these-3-runs-and-determine-the-optimal-value-for-n" class="section level3">
<h3 class="hasAnchor">
<a href="#use-radstackshelpr-to-visualize-the-output-of-these-3-runs-and-determine-the-optimal-value-for-n" class="anchor"></a>Use RADstackshelpR to visualize the output of these 3 runs and determine the optimal value for ‘n’.</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#running in R</span>
<span class="co">#optimize n</span>
<span class="va">n.out</span><span class="op">&lt;-</span><span class="fu"><a href="reference/optimize_n.html">optimize_n</a></span><span class="op">(</span>nequalsMminus1<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/n1.vcf"</span>,
           nequalsM<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/n2.vcf"</span>,
           nequalsMplus1<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/n3.vcf"</span><span class="op">)</span>

<span class="co">##Assigning the output of this function to the variable 'n.out' should generate a single object of class 'data.frame' showing the number of SNPs and loci retained across filtering levels for each value of n.</span>

<span class="co">#visualize the effect of varying n on the number of SNPs retained</span>
<span class="fu"><a href="reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 7420</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-9-1.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying n on the number of polymorphic loci retained</span>
<span class="fu"><a href="reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 3230</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-9-2.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div id="if-you-would-like-you-can-make-a-single-figure-simultaneously-illustrating-the-optimization-process-of-all-three-key-parameters" class="section level4">
<h4 class="hasAnchor">
<a href="#if-you-would-like-you-can-make-a-single-figure-simultaneously-illustrating-the-optimization-process-of-all-three-key-parameters" class="anchor"></a>If you would like, you can make a single figure simultaneously illustrating the optimization process of all three key parameters</h4>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#running in R</span>
<span class="co">#load gridExtra package to combine ggplot visualizations</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>

<span class="co">#combine all of these prior visulizations in a single list</span>
<span class="va">gl</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="reference/vis_depth.html">vis_depth</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span><span class="op">)</span>
<span class="co">#&gt; [1] "Visualize how different values of m affect average depth in each sample"</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">5</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">6</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>

<span class="co">#visualize each item of the list as part of a single grid</span>
<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">gl</span>, widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span>,
  layout_matrix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">3</span><span class="op">)</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">4</span>,<span class="fl">4</span>,<span class="fl">5</span>,<span class="fl">5</span>,<span class="fl">5</span><span class="op">)</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">6</span>,<span class="fl">6</span>,<span class="fl">7</span>,<span class="fl">7</span>,<span class="fl">7</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; Picking joint bandwidth of 9.53</span>
<span class="co">#&gt; Picking joint bandwidth of 7190</span>
<span class="co">#&gt; Picking joint bandwidth of 3420</span>
<span class="co">#&gt; Picking joint bandwidth of 6090</span>
<span class="co">#&gt; Picking joint bandwidth of 2920</span>
<span class="co">#&gt; Picking joint bandwidth of 7420</span>
<span class="co">#&gt; Picking joint bandwidth of 3230</span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-10-1.png" width="120%" height="120%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">#Note: if you are an R whiz and prefer to make your own visualizations, the functions starting with 'optimize_' are designed</span>
<span class="co">#to be modular, and return to you a list of 'data.frame' objects which you can use to make your own custom visualizations.</span></code></pre></div>
<p>You now have a de novo-assembled, parameter-optimized, unfiltered SNP dataset for your organism, and a single cohesive figure documenting the filtering process, which you can stick in your paper’s supplement or host on your own repository. The next step is to filter this vcf file using the SNPfiltR package, which is specifically designed to pick up where this pipeline leaves off, or with any other variant filtering program of your choice.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Devon DeRaad <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-3105-985X" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
