<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Optimize denovo STACKS pipeline for Hipposideros bats â€¢ RADstackshelpR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Optimize denovo STACKS pipeline for Hipposideros bats">
<meta property="og:description" content="RADstackshelpR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RADstackshelpR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/optimize_denovo.html">Optimize denovo STACKS pipeline for Hipposideros bats</a>
    </li>
    <li>
      <a href="../articles/reproducible.vignette.html">Vignette Title</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Optimize denovo STACKS pipeline for Hipposideros bats</h1>
            
      
      
      <div class="hidden name"><code>optimize_denovo.Rmd</code></div>

    </div>

    
    
<div id="this-is-a-real-example-of-using-the-radstackshelpr-package-to-facilitate-parameter-optimization-on-an-empirical-dataset--bash-code-used-to-run-the-stacks-pipeline-was-executed-via-submit-script-on-the-ku-high-performance-computing-cluster-" class="section level5">
<h5 class="hasAnchor">
<a href="#this-is-a-real-example-of-using-the-radstackshelpr-package-to-facilitate-parameter-optimization-on-an-empirical-dataset--bash-code-used-to-run-the-stacks-pipeline-was-executed-via-submit-script-on-the-ku-high-performance-computing-cluster-" class="anchor"></a>This is a real example of using the RADstackshelpR package to facilitate parameter optimization on an empirical dataset. Bash code used to run the <a href="https://catchenlab.life.illinois.edu/stacks/">STACKS</a> pipeline was executed via submit script on the KU high-performance computing cluster.</h5>
<div class="sourceCode"><pre class="downlit">
<span class="co">#load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RADstackshelpR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span></pre></div>
</div>
<div id="step-1-demultiplex-each-sample-using-process_radtags-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-1-demultiplex-each-sample-using-process_radtags-" class="anchor"></a>Step 1: Demultiplex each sample using 'process_radtags'.</h2>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">/home/path/to/stacks-2.41/process_radtags</span> -p .  -o . -b plate.1.barcodes.txt -e ndeI -r -c -q</code></pre></div>
</div>
<div id="step-2-samples-with-a-demultiplexed-file-size-of-2-megabytes-simply-dont-have-enough-information-to-be-relevant-downstream-and-can-be-removed-right-away--note-this-will-help-ensure-that-there-are-enough-loci-retained-at-an-80-completeness-cutoff-to-make-informed-optimization-decisions" class="section level2">
<h2 class="hasAnchor">
<a href="#step-2-samples-with-a-demultiplexed-file-size-of-2-megabytes-simply-dont-have-enough-information-to-be-relevant-downstream-and-can-be-removed-right-away--note-this-will-help-ensure-that-there-are-enough-loci-retained-at-an-80-completeness-cutoff-to-make-informed-optimization-decisions" class="anchor"></a>Step 2: Samples with a demultiplexed file size of &lt; 2 megabytes simply don't have enough information to be relevant downstream, and can be removed right away. (Note: this will help ensure that there are enough loci retained at an 80% completeness cutoff to make informed optimization decisions)</h2>
</div>
<div id="step-3-iterate-over-values-of-m-ranging-from-3-7-while-leaving-all-other-parameters-at-default-values-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-3-iterate-over-values-of-m-ranging-from-3-7-while-leaving-all-other-parameters-at-default-values-" class="anchor"></a>Step 3: Iterate over values of 'm' ranging from 3-7, while leaving all other parameters at default values.</h2>
<div id="bash-code-to-execute-this" class="section level3">
<h3 class="hasAnchor">
<a href="#bash-code-to-execute-this" class="anchor"></a>Bash code to execute this:</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">#!/bin/sh</span>
<span class="co">#</span>
<span class="co">#SBATCH --job-name=denovo.rad.hippo              # Job Name</span>
<span class="co">#SBATCH --nodes=1             # nodes</span>
<span class="co">#SBATCH --cpus-per-task=15               # CPU allocation per Task</span>
<span class="co">#SBATCH --partition=bi            # Name of the Slurm partition used</span>
<span class="co">#SBATCH --chdir=/home/d669d153/scratch/diadema.dinops   # Set working d$</span>
<span class="co">#SBATCH --mem-per-cpu=1gb            # memory requested</span>
<span class="co">#SBATCH --time=10000</span>

<span class="va">files=</span><span class="st">"908108_H_diadema_Gatokae</span>
<span class="st">908150_H_dinops_Guadalcanal</span>
<span class="st">908151_H_diadema_Guadalcanal</span>
<span class="st">908152_H_diadema_Guadalcanal</span>
<span class="st">908153a_H_dinops_Guadalcanal</span>
<span class="st">908154_H_dinops_Guadalcanal</span>
<span class="st">908155_H_dinops_Guadalcanal</span>
<span class="st">908156_H_diadema_Guadalcanal</span>
<span class="st">908208_H_diadema_Guadalcanal</span>
<span class="st">KVO150_H_diadema_Isabel</span>
<span class="st">KVO168_H_diadema_Isabel</span>
<span class="st">KVO169_H_diadema_Isabel</span>
<span class="st">KVO170_H_diadema_Isabel</span>
<span class="st">KVO171_H_diadema_Isabel</span>
<span class="st">KVO172_H_diadema_Isabel</span>
<span class="st">KVO242_H_dinops_Isabel</span>
<span class="st">KVO243_H_dinops_Isabel</span>
<span class="st">KVO244_H_dinops_Isabel</span>
<span class="st">KVO245_H_dinops_Isabel</span>
<span class="st">KVO246_H_dinops_Isabel</span>
<span class="st">KVO248_H_dinops_Isabel</span>
<span class="st">THL1156_H_demissus_Makira</span>
<span class="st">THL1172_H_dinops_Guadalcanal</span>
<span class="st">THL1173_H_dinops_Guadalcanal</span>
<span class="st">THL17193_H_diadema_Ngella</span>
<span class="st">THL17194_H_diadema_Ngell</span>
<span class="st">THL17195_H_diadema_Ngella</span>
<span class="st">THL17197_H_diadema_Ngella</span>
<span class="st">THL17198_H_diadema_Ngella</span>
<span class="st">THL17199_H_diadema_Ngell</span>
<span class="st">WD1705_H_diadema_E_New_Britain</span>
<span class="st">WD2047_H_diadema_Simbu_Prov</span>
<span class="st">WD2074_H_diadema_Gulf_Prov"</span>

<span class="co"># Build loci de novo in each sample for the single-end reads only.</span>
<span class="co"># -M â€” Maximum distance (in nucleotides) allowed between stacks (default 2).</span>
<span class="co"># -m â€” Minimum depth of coverage required to create a stack (default 3).</span>
<span class="co">#here, we will vary m from 3-7, and leave all other paramaters default</span>

<span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{3..7}</span>
<span class="kw">do</span>
<span class="co">#create a directory to hold this unique iteration:</span>
<span class="fu">mkdir</span> stacks_m<span class="va">$i</span>
<span class="co">#run ustacks with m equal to the current iteration (3-7) for each sample</span>
<span class="va">id=</span>1
<span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span>
<span class="kw">do</span>
    <span class="ex">/home/home/d669d153/work/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_m<span class="va">$i</span> -i <span class="va">$id</span> -m <span class="va">$i</span> -p 15
    <span class="bu">let</span> <span class="st">"id+=1"</span>
<span class="kw">done</span>
<span class="co">## Run cstacks to compile stacks between samples. Popmap is a file in working directory called 'pipeline_popmap.txt'</span>
<span class="ex">/home/home/d669d153/work/stacks-2.41/cstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="co">## Run sstacks. Match all samples supplied in the population map against the catalog.</span>
<span class="ex">/home/home/d669d153/work/stacks-2.41/sstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="co">## Run tsv2bam to transpose the data so it is stored by locus, instead of by sample.</span>
<span class="ex">/home/home/d669d153/work/stacks-2.41/tsv2bam</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="co">## Run gstacks: build a paired-end contig from the metapopulation data (if paired-reads provided),</span>
<span class="co">## align reads per sample, call variant sites in the population, genotypes in each individual.</span>
<span class="ex">/home/home/d669d153/work/stacks-2.41/gstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="co">## Run populations completely unfiltered and output unfiltered vcf, for input to the RADstackshelpR package</span>
<span class="ex">/home/home/d669d153/work/stacks-2.41/populations</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15
<span class="kw">done</span></code></pre></div>
</div>
</div>
<div id="step-4-visualize-the-output-of-these-5-runs-and-determine-the-optimal-value-for-m-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-4-visualize-the-output-of-these-5-runs-and-determine-the-optimal-value-for-m-" class="anchor"></a>Step 4: Visualize the output of these 5 runs and determine the optimal value for m.</h2>
<div class="sourceCode"><pre class="downlit">
<span class="co">#optimize_m function will generate summary stats on your 5 iterative runs</span>
<span class="co">#input can be full path to each file or just the file name if your working directory contains the files</span>
<span class="va">m.out</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/optimize_m.html">optimize_m</a></span><span class="op">(</span>m3<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_3.vcf"</span>,
           m4<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_4.vcf"</span>,
           m5<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_5.vcf"</span>,
           m6<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_6.vcf"</span>,
           m7<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/m_7.vcf"</span><span class="op">)</span>

<span class="co">#visualize the effect of varying m on the depth of each sample</span>
<span class="fu"><a href="../reference/vis_depth.html">vis_depth</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span><span class="op">)</span>
<span class="co">#&gt; [1] "Visualize how different values of m affect average depth in each sample"</span>
<span class="co">#&gt; Picking joint bandwidth of 9.53</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-5-1.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode"><pre class="downlit">
<span class="co">#visualize the effect of varying m on the number of SNPs retained</span>
<span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 7190</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-5-2.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode"><pre class="downlit">

<span class="co">#visualize the effect of varying m on the number of loci retained</span>
<span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 3420</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-5-3.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode"><pre class="downlit">

<span class="co">#3 is the optimal m value, and will be used next to optimize M</span></pre></div>
</div>
<div id="step-5-iterate-over-values-of-m-ranging-from-1-8-setting-m-to-the-optimal-value-here-3-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-5-iterate-over-values-of-m-ranging-from-1-8-setting-m-to-the-optimal-value-here-3-" class="anchor"></a>Step 5: Iterate over values of M ranging from 1-8, setting m to the optimal value (here 3).</h2>
<div id="this-code-follows-the-bash-chunk-above-utilizing-the-same-variable-files" class="section level3">
<h3 class="hasAnchor">
<a href="#this-code-follows-the-bash-chunk-above-utilizing-the-same-variable-files" class="anchor"></a>This code follows the bash chunk above, utilizing the same variable $files</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># -M â€” Maximum distance (in nucleotides) allowed between stacks (default 2).</span>
<span class="co"># -m â€” Minimum depth of coverage required to create a stack (default 3).</span>
<span class="co">#here, vary M from 1-8, and set m to the optimized value based on prior visualizations (here 3)</span>
<span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{1..8}</span>
<span class="kw">do</span>
<span class="co">#create a directory to hold this unique iteration:</span>
<span class="fu">mkdir</span> stacks_bigM<span class="va">$i</span>
<span class="co">#run ustacks with M equal to the current iteration (1-8) for each sample, and m set to the optimized value (here, m=3)</span>
<span class="va">id=</span>1
<span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span>
<span class="kw">do</span>
    <span class="ex">/home/d669d153/work/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_bigM<span class="va">$i</span> -i <span class="va">$id</span> -m 3 -M <span class="va">$i</span> -p 15
    <span class="bu">let</span> <span class="st">"id+=1"</span>
<span class="kw">done</span>
<span class="ex">/home/d669d153/work/stacks-2.41/cstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/d669d153/work/stacks-2.41/sstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/d669d153/work/stacks-2.41/tsv2bam</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/d669d153/work/stacks-2.41/gstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/d669d153/work/stacks-2.41/populations</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15
<span class="kw">done</span></code></pre></div>
</div>
</div>
<div id="step-6-visualize-the-output-of-these-8-runs-and-determine-the-optimal-value-for-m-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-6-visualize-the-output-of-these-8-runs-and-determine-the-optimal-value-for-m-" class="anchor"></a>Step 6: Visualize the output of these 8 runs and determine the optimal value for M.</h2>
<div class="sourceCode"><pre class="downlit">
<span class="co">#optimize M</span>
<span class="va">M.out</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/optimize_bigM.html">optimize_bigM</a></span><span class="op">(</span>M1<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M1.vcf"</span>,
           M2<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M2.vcf"</span>,
           M3<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M3.vcf"</span>,
           M4<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M4.vcf"</span>,
           M5<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M5.vcf"</span>,
           M6<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M6.vcf"</span>,
           M7<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M7.vcf"</span>,
           M8<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/M8.vcf"</span><span class="op">)</span>

<span class="co">#visualize the effect of varying M on the number of SNPs retained</span>
<span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 6090</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-7-1.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode"><pre class="downlit">

<span class="co">#visualize the effect of varying M on the number of polymorphic loci retained</span>
<span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 2920</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-7-2.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
</div>
<div id="step-7-iterate-over-values-of-n-ranging-from-m-1-m-m1-setting-m-and-m-to-the-optimal-values-here-3-and-2-respectively-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-7-iterate-over-values-of-n-ranging-from-m-1-m-m1-setting-m-and-m-to-the-optimal-values-here-3-and-2-respectively-" class="anchor"></a>Step 7: Iterate over values of n ranging from M-1, M, M+1, setting m and M to the optimal values (here 3 and 2, respectively).</h2>
<div id="this-code-chunk-runs-the-last-three-stacks-iterations-again-re-using-the-variable-files" class="section level3">
<h3 class="hasAnchor">
<a href="#this-code-chunk-runs-the-last-three-stacks-iterations-again-re-using-the-variable-files" class="anchor"></a>This code chunk runs the last three Stacks iterations, again re-using the variable $files</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># -n â€” Number of mismatches allowed between sample loci when build the catalog (default 1).</span>
<span class="co">#here, vary 'n' across M-1, M, and M+1 (because my optimized 'M' value = 2, I will iterate over 1, 2, and 3 here), with 'm' and 'M' set to the optimized value based on prior visualizations (here 'm' = 3, and 'M'=2).</span>
<span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{1..3}</span>
<span class="kw">do</span>
<span class="co">#create a directory to hold this unique iteration:</span>
<span class="fu">mkdir</span> stacks_n<span class="va">$i</span>
<span class="co">#run ustacks with n equal to the current iteration (1-3) for each sample, m = 3, and M=2</span>
<span class="va">id=</span>1
<span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span>
<span class="kw">do</span>
    <span class="ex">/home/d669d153/work/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_n<span class="va">$i</span> -i <span class="va">$id</span> -m 3 -M 2 -p 15
    <span class="bu">let</span> <span class="st">"id+=1"</span>
<span class="kw">done</span>
<span class="ex">/home/d669d153/work/stacks-2.41/cstacks</span> -n <span class="va">$i</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/d669d153/work/stacks-2.41/sstacks</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/d669d153/work/stacks-2.41/tsv2bam</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/d669d153/work/stacks-2.41/gstacks</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/d669d153/work/stacks-2.41/populations</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15
<span class="kw">done</span></code></pre></div>
</div>
</div>
<div id="step-8-visualize-the-output-of-these-3-runs-and-determine-the-optimal-value-for-n-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-8-visualize-the-output-of-these-3-runs-and-determine-the-optimal-value-for-n-" class="anchor"></a>Step 8: Visualize the output of these 3 runs and determine the optimal value for n.</h2>
<div class="sourceCode"><pre class="downlit">
<span class="co">#optimize n</span>
<span class="va">n.out</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/optimize_n.html">optimize_n</a></span><span class="op">(</span>nequalsMminus1<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/n1.vcf"</span>,
           nequalsM<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/n2.vcf"</span>,
           nequalsMplus1<span class="op">=</span><span class="st">"/Users/devder/Desktop/hipposideros/n3.vcf"</span><span class="op">)</span>

<span class="co">#visualize the effect of varying n on the number of SNPs retained</span>
<span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 7420</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-9-1.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode"><pre class="downlit">

<span class="co">#visualize the effect of varying n on the number of polymorphic loci retained</span>
<span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 3230</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-9-2.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
</div>
<div id="finally-make-a-single-figure-showing-the-optimization-process-of-all-three-parameters" class="section level1">
<h1 class="hasAnchor">
<a href="#finally-make-a-single-figure-showing-the-optimization-process-of-all-three-parameters" class="anchor"></a>Finally, make a single figure showing the optimization process of all three parameters</h1>
<div class="sourceCode"><pre class="downlit">
<span class="va">gl</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/vis_depth.html">vis_depth</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span><span class="op">)</span>
<span class="co">#&gt; [1] "Visualize how different values of m affect average depth in each sample"</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">5</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">6</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="va">gl</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>

<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span>
  grobs <span class="op">=</span> <span class="va">gl</span>,
  widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span>,
  layout_matrix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">3</span><span class="op">)</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">4</span>,<span class="fl">4</span>,<span class="fl">5</span>,<span class="fl">5</span>,<span class="fl">5</span><span class="op">)</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">6</span>,<span class="fl">6</span>,<span class="fl">7</span>,<span class="fl">7</span>,<span class="fl">7</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; Picking joint bandwidth of 9.53</span>
<span class="co">#&gt; Picking joint bandwidth of 7190</span>
<span class="co">#&gt; Picking joint bandwidth of 3420</span>
<span class="co">#&gt; Picking joint bandwidth of 6090</span>
<span class="co">#&gt; Picking joint bandwidth of 2920</span>
<span class="co">#&gt; Picking joint bandwidth of 7420</span>
<span class="co">#&gt; Picking joint bandwidth of 3230</span></pre></div>
<p><img src="optimize_denovo_files/figure-html/unnamed-chunk-10-1.png" width="100%" height="100%" style="display: block; margin: auto;"></p>
<div id="i-now-have-a-vcf-file-containing-unfiltered-snps-for-all-samples-built-with-the-optimal-parameters-for-this-dataset-m-3-m-2-and-n3-" class="section level5">
<h5 class="hasAnchor">
<a href="#i-now-have-a-vcf-file-containing-unfiltered-snps-for-all-samples-built-with-the-optimal-parameters-for-this-dataset-m-3-m-2-and-n3-" class="anchor"></a>I now have a vcf file containing unfiltered SNPs for all samples, built with the optimal parameters for this dataset, 'm' = 3, 'M' = 2, and 'n'=3.</h5>
</div>
<div id="i-will-now-filter-this-vcf-file-using-the-snpfiltr-r-package-to-generate-a-set-of-high-quality-snps-for-downstream-phylogeneticpopulation-genetic-analyses-" class="section level5">
<h5 class="hasAnchor">
<a href="#i-will-now-filter-this-vcf-file-using-the-snpfiltr-r-package-to-generate-a-set-of-high-quality-snps-for-downstream-phylogeneticpopulation-genetic-analyses-" class="anchor"></a>I will now filter this vcf file using the SNPfiltR R package, to generate a set of high quality SNPs for downstream phylogenetic/population genetic analyses.</h5>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
