<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Quality Control Vignette • RADstackshelpR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Quality Control Vignette">
<meta property="og:description" content="RADstackshelpR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-TB39NWWF3X"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-TB39NWWF3X');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RADstackshelpR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/brown.trout.validation.html">Brown Trout Validation</a>
    </li>
    <li>
      <a href="../articles/optimize_denovo.html">Optimize denovo STACKS pipeline for Hipposideros bats</a>
    </li>
    <li>
      <a href="../articles/quality.control.vignette.html">Quality Control Vignette</a>
    </li>
    <li>
      <a href="../articles/reproducible.vignette.html">Reproducible example</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="quality.control.vignette_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Quality Control Vignette</h1>
                        <h4 class="author">Devon DeRaad</h4>
            
            <h4 class="date">24 September 2021</h4>
      
      
      <div class="hidden name"><code>quality.control.vignette.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/knitr/">knitr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="co">#install.packages("fastqcr")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.sthda.com/english/rpkgs/fastqcr/">fastqcr</a></span><span class="op">)</span>
<span class="co">#must run this if fastqc is not already installed locally</span>
<span class="co">#fastqc_install()</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">###ONLY THIS CHUNK REQUIRES MODIFICATION###</span>
<span class="co">###assign your directory locations here:</span>

<span class="co">#specify full path to directory containing a .fastq.gz file for each sample</span>
<span class="va">fq.dir</span><span class="op">&lt;-</span><span class="st">"/home/d669d153/work/dia.din/fq"</span>

<span class="co">#specify full path to the output directory where you want </span>
<span class="va">qc.dir</span><span class="op">&lt;-</span><span class="st">"~/Downloads/qc"</span>

<span class="co">#run fastqc on all .fastq.gz files, through r</span>
<span class="co">#only needs to be run once, if tweaking downstream visualizations, comment out this step</span>
<span class="co">#fastqc(fq.dir = fq.dir, # FASTQ files directory</span>
<span class="co">#       qc.dir = qc.dir, # Results directory</span>
<span class="co">#       threads = 4                    # Number of threads</span>
<span class="co">#       )</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># List of files in the output directory to ensure fastqc worked</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">qc.dir</span><span class="op">)</span></code></pre></div>
<p>[1] “908108_H_diadema_Gatokae_fastqc.html”<br>
[2] “908108_H_diadema_Gatokae_fastqc.zip”<br>
[3] “908150_H_dinops_Guadalcanal_fastqc.html”<br>
[4] “908150_H_dinops_Guadalcanal_fastqc.zip”<br>
[5] “908151_H_diadema_Guadalcanal_fastqc.html”<br>
[6] “908151_H_diadema_Guadalcanal_fastqc.zip”<br>
[7] “908152_H_diadema_Guadalcanal_fastqc.html”<br>
[8] “908152_H_diadema_Guadalcanal_fastqc.zip”<br>
[9] “908153a_H_dinops_Guadalcanal_fastqc.html”<br>
[10] “908153a_H_dinops_Guadalcanal_fastqc.zip”<br>
[11] “908154_H_dinops_Guadalcanal_fastqc.html”<br>
[12] “908154_H_dinops_Guadalcanal_fastqc.zip”<br>
[13] “908155_H_dinops_Guadalcanal_fastqc.html”<br>
[14] “908155_H_dinops_Guadalcanal_fastqc.zip”<br>
[15] “908156_H_diadema_Guadalcanal_fastqc.html”<br>
[16] “908156_H_diadema_Guadalcanal_fastqc.zip”<br>
[17] “908208_H_diadema_Guadalcanal_fastqc.html”<br>
[18] “908208_H_diadema_Guadalcanal_fastqc.zip”<br>
[19] “JM19686_H_diadema_Choiseul_fastqc.html”<br>
[20] “JM19686_H_diadema_Choiseul_fastqc.zip”<br>
[21] “KVO150_H_diadema_Isabel_fastqc.html”<br>
[22] “KVO150_H_diadema_Isabel_fastqc.zip”<br>
[23] “KVO168_H_diadema_Isabel_fastqc.html”<br>
[24] “KVO168_H_diadema_Isabel_fastqc.zip”<br>
[25] “KVO169_H_diadema_Isabel_fastqc.html”<br>
[26] “KVO169_H_diadema_Isabel_fastqc.zip”<br>
[27] “KVO170_H_diadema_Isabel_fastqc.html”<br>
[28] “KVO170_H_diadema_Isabel_fastqc.zip”<br>
[29] “KVO171_H_diadema_Isabel_fastqc.html”<br>
[30] “KVO171_H_diadema_Isabel_fastqc.zip”<br>
[31] “KVO172_H_diadema_Isabel_fastqc.html”<br>
[32] “KVO172_H_diadema_Isabel_fastqc.zip”<br>
[33] “KVO242_H_dinops_Isabel_fastqc.html”<br>
[34] “KVO242_H_dinops_Isabel_fastqc.zip”<br>
[35] “KVO243_H_dinops_Isabel_fastqc.html”<br>
[36] “KVO243_H_dinops_Isabel_fastqc.zip”<br>
[37] “KVO244_H_dinops_Isabel_fastqc.html”<br>
[38] “KVO244_H_dinops_Isabel_fastqc.zip”<br>
[39] “KVO245_H_dinops_Isabel_fastqc.html”<br>
[40] “KVO245_H_dinops_Isabel_fastqc.zip”<br>
[41] “KVO246_H_dinops_Isabel_fastqc.html”<br>
[42] “KVO246_H_dinops_Isabel_fastqc.zip”<br>
[43] “KVO248_H_dinops_Isabel_fastqc.html”<br>
[44] “KVO248_H_dinops_Isabel_fastqc.zip”<br>
[45] “KVO249_Hipposiderous_sp_Isabel_fastqc.html” [46] “KVO249_Hipposiderous_sp_Isabel_fastqc.zip”<br>
[47] “KVO250_Hipposiderous_sp_Isabel_fastqc.html” [48] “KVO250_Hipposiderous_sp_Isabel_fastqc.zip”<br>
[49] “KVO251_Hipposiderous_sp_Rendova_fastqc.html” [50] “KVO251_Hipposiderous_sp_Rendova_fastqc.zip” [51] “THL1048_H_dinops_Guadalcanal_fastqc.html”<br>
[52] “THL1048_H_dinops_Guadalcanal_fastqc.zip”<br>
[53] “THL1120_H_dinops_Gatokae_fastqc.html”<br>
[54] “THL1120_H_dinops_Gatokae_fastqc.zip”<br>
[55] “THL1121_H_dinops_Gatokae_fastqc.html”<br>
[56] “THL1121_H_dinops_Gatokae_fastqc.zip”<br>
[57] “THL1122_H_dinops_Gatokae_fastqc.html”<br>
[58] “THL1122_H_dinops_Gatokae_fastqc.zip”<br>
[59] “THL1154_H_demissus_Makira_fastqc.html”<br>
[60] “THL1154_H_demissus_Makira_fastqc.zip”<br>
[61] “THL1156_H_demissus_Makira_fastqc.html”<br>
[62] “THL1156_H_demissus_Makira_fastqc.zip”<br>
[63] “THL1167_H_diadema_Guadalcanal_fastqc.html”<br>
[64] “THL1167_H_diadema_Guadalcanal_fastqc.zip”<br>
[65] “THL1172_H_dinops_Guadalcanal_fastqc.html”<br>
[66] “THL1172_H_dinops_Guadalcanal_fastqc.zip”<br>
[67] “THL1173_H_dinops_Guadalcanal_fastqc.html”<br>
[68] “THL1173_H_dinops_Guadalcanal_fastqc.zip”<br>
[69] “THL1221_H_diadema_Gatokae_fastqc.html”<br>
[70] “THL1221_H_diadema_Gatokae_fastqc.zip”<br>
[71] “THL1223_H_dinops_Guadalcanal_fastqc.html”<br>
[72] “THL1223_H_dinops_Guadalcanal_fastqc.zip”<br>
[73] “THL17193_H_diadema_Ngella_fastqc.html”<br>
[74] “THL17193_H_diadema_Ngella_fastqc.zip”<br>
[75] “THL17194_H_diadema_Ngella_fastqc.html”<br>
[76] “THL17194_H_diadema_Ngella_fastqc.zip”<br>
[77] “THL17195_H_diadema_Ngella_fastqc.html”<br>
[78] “THL17195_H_diadema_Ngella_fastqc.zip”<br>
[79] “THL17197_H_diadema_Ngella_fastqc.html”<br>
[80] “THL17197_H_diadema_Ngella_fastqc.zip”<br>
[81] “THL17198_H_diadema_Ngella_fastqc.html”<br>
[82] “THL17198_H_diadema_Ngella_fastqc.zip”<br>
[83] “THL17199_H_diadema_Ngella_fastqc.html”<br>
[84] “THL17199_H_diadema_Ngella_fastqc.zip”<br>
[85] “WD1705_H_diadema_E_New_Britain_fastqc.html” [86] “WD1705_H_diadema_E_New_Britain_fastqc.zip”<br>
[87] “WD2047_H_diadema_Simbu_Prov_fastqc.html”<br>
[88] “WD2047_H_diadema_Simbu_Prov_fastqc.zip”<br>
[89] “WD2074_H_diadema_Gulf_Prov_fastqc.html”<br>
[90] “WD2074_H_diadema_Gulf_Prov_fastqc.zip”</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#create a character vector where each value is the full path to the .zip created by fastqc() for a given sample</span>
<span class="co">#samps&lt;-list.files("/home/d669d153/work/dia.din/qc", full.names = T, pattern = "*.zip")</span>
<span class="va">samps</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">qc.dir</span>, full.names <span class="op">=</span> <span class="cn">T</span>, pattern <span class="op">=</span> <span class="st">"*.zip"</span><span class="op">)</span>

<span class="co">#plot qc test results for each sample</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">samps</span><span class="op">)</span><span class="op">{</span>
  <span class="co">#read info for given sample from the .zip file generated in the previous step</span>
  <span class="va">samp.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fastqcr/man/qc_read.html">qc_read</a></span><span class="op">(</span><span class="va">i</span><span class="op">)</span>
  <span class="co">#open blank list to hold qc visualizations for the given sample</span>
  <span class="va">plot</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>
  <span class="co">#do qc for the given sample</span>
  <span class="va">plot</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/fastqcr/man/qc_plot.html">qc_plot</a></span><span class="op">(</span><span class="va">samp.info</span>, <span class="st">"Basic statistics"</span><span class="op">)</span>
  <span class="va">plot</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/fastqcr/man/qc_plot.html">qc_plot</a></span><span class="op">(</span><span class="va">samp.info</span>, <span class="st">"Per sequence quality scores"</span><span class="op">)</span>
  <span class="va">plot</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/fastqcr/man/qc_plot.html">qc_plot</a></span><span class="op">(</span><span class="va">samp.info</span>, <span class="st">"Sequence duplication levels"</span><span class="op">)</span>
  <span class="co">#visualize tables</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"QC results for sample "</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">".*/"</span>, <span class="st">""</span>, <span class="va">i</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">'\n'</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">plot</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">'\n'</span><span class="op">)</span>

  <span class="co">#visualize plots</span>
  <span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">plot</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>,<span class="va">plot</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>,
               ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
  
  <span class="co">#clear plot to hold info for next sample</span>
  <span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">plot</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>[1] “QC results for sample 908108_H_diadema_Gatokae_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908108_H_diadema_Gatokae.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">454706</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-1.png" width="100%" height="100%">[1] “QC results for sample 908150_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908150_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">120063</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">38</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-2.png" width="100%" height="100%">[1] “QC results for sample 908151_H_diadema_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908151_H_diadema_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">3174216</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">38</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-3.png" width="100%" height="100%">[1] “QC results for sample 908152_H_diadema_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908152_H_diadema_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1388568</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-4.png" width="100%" height="100%">[1] “QC results for sample 908153a_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908153a_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">7300968</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-5.png" width="100%" height="100%">[1] “QC results for sample 908154_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908154_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">519876</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-6.png" width="100%" height="100%">[1] “QC results for sample 908155_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908155_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">989397</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-7.png" width="100%" height="100%">[1] “QC results for sample 908156_H_diadema_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908156_H_diadema_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">4264843</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-8.png" width="100%" height="100%">[1] “QC results for sample 908208_H_diadema_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">908208_H_diadema_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">769417</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-9.png" width="100%" height="100%">[1] “QC results for sample JM19686_H_diadema_Choiseul_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">JM19686_H_diadema_Choiseul.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1029</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">34</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-10.png" width="100%" height="100%">[1] “QC results for sample KVO150_H_diadema_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO150_H_diadema_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">34555</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-11.png" width="100%" height="100%">[1] “QC results for sample KVO168_H_diadema_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO168_H_diadema_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">262565</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-12.png" width="100%" height="100%">[1] “QC results for sample KVO169_H_diadema_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO169_H_diadema_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">499713</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-13.png" width="100%" height="100%">[1] “QC results for sample KVO170_H_diadema_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO170_H_diadema_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">825167</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-14.png" width="100%" height="100%">[1] “QC results for sample KVO171_H_diadema_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO171_H_diadema_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">271596</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-15.png" width="100%" height="100%">[1] “QC results for sample KVO172_H_diadema_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO172_H_diadema_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">98438</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-16.png" width="100%" height="100%">[1] “QC results for sample KVO242_H_dinops_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO242_H_dinops_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1350278</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-17.png" width="100%" height="100%">[1] “QC results for sample KVO243_H_dinops_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO243_H_dinops_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1797464</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-18.png" width="100%" height="100%">[1] “QC results for sample KVO244_H_dinops_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO244_H_dinops_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">3002231</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-19.png" width="100%" height="100%">[1] “QC results for sample KVO245_H_dinops_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO245_H_dinops_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">4187253</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-20.png" width="100%" height="100%">[1] “QC results for sample KVO246_H_dinops_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO246_H_dinops_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">2153670</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-21.png" width="100%" height="100%">[1] “QC results for sample KVO248_H_dinops_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO248_H_dinops_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1799491</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-22.png" width="100%" height="100%">[1] “QC results for sample KVO249_Hipposiderous_sp_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO249_Hipposiderous_sp_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">2164473</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-23.png" width="100%" height="100%">[1] “QC results for sample KVO250_Hipposiderous_sp_Isabel_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO250_Hipposiderous_sp_Isabel.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">215420</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-24.png" width="100%" height="100%">[1] “QC results for sample KVO251_Hipposiderous_sp_Rendova_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">KVO251_Hipposiderous_sp_Rendova.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1739541</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-25.png" width="100%" height="100%">[1] “QC results for sample THL1048_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1048_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">4848</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">35</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-26.png" width="100%" height="100%">[1] “QC results for sample THL1120_H_dinops_Gatokae_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1120_H_dinops_Gatokae.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">3212</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-27.png" width="100%" height="100%">[1] “QC results for sample THL1121_H_dinops_Gatokae_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1121_H_dinops_Gatokae.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">6874</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">35</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-28.png" width="100%" height="100%">[1] “QC results for sample THL1122_H_dinops_Gatokae_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1122_H_dinops_Gatokae.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">288</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-29.png" width="100%" height="100%">[1] “QC results for sample THL1154_H_demissus_Makira_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1154_H_demissus_Makira.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">5201</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-30.png" width="100%" height="100%">[1] “QC results for sample THL1156_H_demissus_Makira_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1156_H_demissus_Makira.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">819353</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-31.png" width="100%" height="100%">[1] “QC results for sample THL1167_H_diadema_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1167_H_diadema_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1379</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-32.png" width="100%" height="100%">[1] “QC results for sample THL1172_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1172_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">173703</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">38</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-33.png" width="100%" height="100%">[1] “QC results for sample THL1173_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1173_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">979903</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-34.png" width="100%" height="100%">[1] “QC results for sample THL1221_H_diadema_Gatokae_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1221_H_diadema_Gatokae.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">43</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">40</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-35.png" width="100%" height="100%">[1] “QC results for sample THL1223_H_dinops_Guadalcanal_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL1223_H_dinops_Guadalcanal.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">6655</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-36.png" width="100%" height="100%">[1] “QC results for sample THL17193_H_diadema_Ngella_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL17193_H_diadema_Ngella.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">3033643</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">38</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-37.png" width="100%" height="100%">[1] “QC results for sample THL17194_H_diadema_Ngella_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL17194_H_diadema_Ngella.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1210618</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-38.png" width="100%" height="100%">[1] “QC results for sample THL17195_H_diadema_Ngella_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL17195_H_diadema_Ngella.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1825457</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-39.png" width="100%" height="100%">[1] “QC results for sample THL17197_H_diadema_Ngella_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL17197_H_diadema_Ngella.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1546118</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-40.png" width="100%" height="100%">[1] “QC results for sample THL17198_H_diadema_Ngella_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL17198_H_diadema_Ngella.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">6607272</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-41.png" width="100%" height="100%">[1] “QC results for sample THL17199_H_diadema_Ngella_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">THL17199_H_diadema_Ngella.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1508601</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-42.png" width="100%" height="100%">[1] “QC results for sample WD1705_H_diadema_E_New_Britain_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">WD1705_H_diadema_E_New_Britain.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">1323647</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">37</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-43.png" width="100%" height="100%">[1] “QC results for sample WD2047_H_diadema_Simbu_Prov_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">WD2047_H_diadema_Simbu_Prov.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">696126</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-44.png" width="100%" height="100%">[1] “QC results for sample WD2074_H_diadema_Gulf_Prov_fastqc.zip”</p>
<table class="table">
<thead><tr class="header">
<th align="left">Measure</th>
<th align="left">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Filename</td>
<td align="left">WD2074_H_diadema_Gulf_Prov.fq.gz</td>
</tr>
<tr class="even">
<td align="left">File type</td>
<td align="left">Conventional base calls</td>
</tr>
<tr class="odd">
<td align="left">Encoding</td>
<td align="left">Sanger / Illumina 1.9</td>
</tr>
<tr class="even">
<td align="left">Total Sequences</td>
<td align="left">945453</td>
</tr>
<tr class="odd">
<td align="left">Sequences flagged as poor quality</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">Sequence length</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">%GC</td>
<td align="left">36</td>
</tr>
</tbody>
</table>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-3-45.png" width="100%" height="100%"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#aggregate the reports by pointing this function to the folder holding output of fastqc()</span>
<span class="co">#qc &lt;- qc_aggregate("/home/d669d153/work/dia.din/qc", progressbar=F)</span>
<span class="va">qc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/fastqcr/man/qc_aggregate.html">qc_aggregate</a></span><span class="op">(</span><span class="va">qc.dir</span>, progressbar <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>

<span class="co">#stats per sample</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/fastqcr/man/qc_aggregate.html">qc_stats</a></span><span class="op">(</span><span class="va">qc</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">sample</th>
<th align="right">pct.dup</th>
<th align="right">pct.gc</th>
<th align="left">tot.seq</th>
<th align="left">seq.length</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">908108_H_diadema_Gatokae.fq.gz</td>
<td align="right">84.98</td>
<td align="right">36</td>
<td align="left">454706</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">908150_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">76.29</td>
<td align="right">38</td>
<td align="left">120063</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">908151_H_diadema_Guadalcanal.fq.gz</td>
<td align="right">94.54</td>
<td align="right">38</td>
<td align="left">3174216</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">908152_H_diadema_Guadalcanal.fq.gz</td>
<td align="right">89.86</td>
<td align="right">37</td>
<td align="left">1388568</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">908153a_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">94.82</td>
<td align="right">37</td>
<td align="left">7300968</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">908154_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">84.86</td>
<td align="right">37</td>
<td align="left">519876</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">908155_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">88.96</td>
<td align="right">37</td>
<td align="left">989397</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">908156_H_diadema_Guadalcanal.fq.gz</td>
<td align="right">94.89</td>
<td align="right">37</td>
<td align="left">4264843</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">908208_H_diadema_Guadalcanal.fq.gz</td>
<td align="right">89.66</td>
<td align="right">37</td>
<td align="left">769417</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">JM19686_H_diadema_Choiseul.fq.gz</td>
<td align="right">18.85</td>
<td align="right">34</td>
<td align="left">1029</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO150_H_diadema_Isabel.fq.gz</td>
<td align="right">58.87</td>
<td align="right">36</td>
<td align="left">34555</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO168_H_diadema_Isabel.fq.gz</td>
<td align="right">80.63</td>
<td align="right">36</td>
<td align="left">262565</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO169_H_diadema_Isabel.fq.gz</td>
<td align="right">86.28</td>
<td align="right">36</td>
<td align="left">499713</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO170_H_diadema_Isabel.fq.gz</td>
<td align="right">89.21</td>
<td align="right">36</td>
<td align="left">825167</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO171_H_diadema_Isabel.fq.gz</td>
<td align="right">81.79</td>
<td align="right">36</td>
<td align="left">271596</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO172_H_diadema_Isabel.fq.gz</td>
<td align="right">70.24</td>
<td align="right">36</td>
<td align="left">98438</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO242_H_dinops_Isabel.fq.gz</td>
<td align="right">91.91</td>
<td align="right">37</td>
<td align="left">1350278</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO243_H_dinops_Isabel.fq.gz</td>
<td align="right">92.18</td>
<td align="right">37</td>
<td align="left">1797464</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO244_H_dinops_Isabel.fq.gz</td>
<td align="right">93.16</td>
<td align="right">36</td>
<td align="left">3002231</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO245_H_dinops_Isabel.fq.gz</td>
<td align="right">94.18</td>
<td align="right">36</td>
<td align="left">4187253</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO246_H_dinops_Isabel.fq.gz</td>
<td align="right">91.61</td>
<td align="right">36</td>
<td align="left">2153670</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO248_H_dinops_Isabel.fq.gz</td>
<td align="right">91.09</td>
<td align="right">36</td>
<td align="left">1799491</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO249_Hipposiderous_sp_Isabel.fq.gz</td>
<td align="right">91.67</td>
<td align="right">36</td>
<td align="left">2164473</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO250_Hipposiderous_sp_Isabel.fq.gz</td>
<td align="right">79.87</td>
<td align="right">36</td>
<td align="left">215420</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">KVO251_Hipposiderous_sp_Rendova.fq.gz</td>
<td align="right">91.80</td>
<td align="right">36</td>
<td align="left">1739541</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1048_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">35.68</td>
<td align="right">35</td>
<td align="left">4848</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1120_H_dinops_Gatokae.fq.gz</td>
<td align="right">10.65</td>
<td align="right">36</td>
<td align="left">3212</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1121_H_dinops_Gatokae.fq.gz</td>
<td align="right">9.18</td>
<td align="right">35</td>
<td align="left">6874</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1122_H_dinops_Gatokae.fq.gz</td>
<td align="right">43.75</td>
<td align="right">36</td>
<td align="left">288</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1154_H_demissus_Makira.fq.gz</td>
<td align="right">15.44</td>
<td align="right">36</td>
<td align="left">5201</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1156_H_demissus_Makira.fq.gz</td>
<td align="right">88.94</td>
<td align="right">36</td>
<td align="left">819353</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1167_H_diadema_Guadalcanal.fq.gz</td>
<td align="right">38.43</td>
<td align="right">37</td>
<td align="left">1379</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1172_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">77.77</td>
<td align="right">38</td>
<td align="left">173703</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1173_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">88.50</td>
<td align="right">37</td>
<td align="left">979903</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1221_H_diadema_Gatokae.fq.gz</td>
<td align="right">39.53</td>
<td align="right">40</td>
<td align="left">43</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1223_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">14.46</td>
<td align="right">36</td>
<td align="left">6655</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL17193_H_diadema_Ngella.fq.gz</td>
<td align="right">94.40</td>
<td align="right">38</td>
<td align="left">3033643</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL17194_H_diadema_Ngella.fq.gz</td>
<td align="right">90.91</td>
<td align="right">37</td>
<td align="left">1210618</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL17195_H_diadema_Ngella.fq.gz</td>
<td align="right">92.44</td>
<td align="right">37</td>
<td align="left">1825457</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL17197_H_diadema_Ngella.fq.gz</td>
<td align="right">91.87</td>
<td align="right">37</td>
<td align="left">1546118</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL17198_H_diadema_Ngella.fq.gz</td>
<td align="right">95.67</td>
<td align="right">37</td>
<td align="left">6607272</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL17199_H_diadema_Ngella.fq.gz</td>
<td align="right">91.37</td>
<td align="right">37</td>
<td align="left">1508601</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">WD1705_H_diadema_E_New_Britain.fq.gz</td>
<td align="right">91.09</td>
<td align="right">37</td>
<td align="left">1323647</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">WD2047_H_diadema_Simbu_Prov.fq.gz</td>
<td align="right">86.17</td>
<td align="right">36</td>
<td align="left">696126</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">WD2074_H_diadema_Gulf_Prov.fq.gz</td>
<td align="right">88.42</td>
<td align="right">36</td>
<td align="left">945453</td>
<td align="left">145</td>
</tr>
</tbody>
</table>
<div id="solid-red-line-median-sample-value" class="section level3">
<h3 class="hasAnchor">
<a href="#solid-red-line-median-sample-value" class="anchor"></a>solid red line = median sample value</h3>
</div>
<div id="dashed-red-line-10-of-median-sample-value" class="section level3">
<h3 class="hasAnchor">
<a href="#dashed-red-line-10-of-median-sample-value" class="anchor"></a>dashed red line = 10% of median sample value</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#save stats info as an object</span>
<span class="va">stats.info</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/fastqcr/man/qc_aggregate.html">qc_stats</a></span><span class="op">(</span><span class="va">qc</span><span class="op">)</span>
<span class="co">#make tot.seq numeric</span>
<span class="va">stats.info</span><span class="op">$</span><span class="va">tot.seq</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">stats.info</span><span class="op">$</span><span class="va">tot.seq</span><span class="op">)</span>

<span class="co">#make histogram of number of sequence reads for each sample</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">stats.info</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, fill<span class="op">=</span><span class="st">"white"</span>, bins<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xintercept<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xintercept<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">*</span><span class="fl">.1</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"red"</span>, lty<span class="op">=</span><span class="fl">14</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Number of sequencing reads"</span><span class="op">)</span></code></pre></div>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-5-1.png" width="65%"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#solid red line = median sample value</span>
<span class="co">#dashed red line = 10% of median sample value</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">stats.info</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, fill<span class="op">=</span><span class="st">"white"</span>, bins<span class="op">=</span><span class="fl">200</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xintercept<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xintercept<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">*</span><span class="fl">.1</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"red"</span>, lty<span class="op">=</span><span class="fl">14</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Number of sequencing reads"</span><span class="op">)</span></code></pre></div>
<p><img src="quality.control.vignette_files/figure-html/unnamed-chunk-5-2.png" width="65%"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#show me the samples that have less than 10% of the number of reads as the median sample from this experiment (these should be dropped immediately)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Median sample contains"</span>, <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">stats.info</span><span class="op">$</span><span class="va">tot.seq</span><span class="op">)</span>, <span class="st">"reads. The following samples contain less than"</span>, <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">stats.info</span><span class="op">$</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">*</span><span class="fl">.1</span>, <span class="st">"reads (10% of the median), and should likely be dropped"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>[1] “Median sample contains 825167 reads. The following samples contain less than 82516.7 reads (10% of the median), and should likely be dropped”</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">stats.info</span><span class="op">[</span><span class="va">stats.info</span><span class="op">$</span><span class="va">tot.seq</span> <span class="op">&lt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">stats.info</span><span class="op">$</span><span class="va">tot.seq</span><span class="op">)</span><span class="op">*</span><span class="fl">.1</span>,<span class="op">]</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">sample</th>
<th align="right">pct.dup</th>
<th align="right">pct.gc</th>
<th align="right">tot.seq</th>
<th align="left">seq.length</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">JM19686_H_diadema_Choiseul.fq.gz</td>
<td align="right">18.85</td>
<td align="right">34</td>
<td align="right">1029</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">KVO150_H_diadema_Isabel.fq.gz</td>
<td align="right">58.87</td>
<td align="right">36</td>
<td align="right">34555</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1048_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">35.68</td>
<td align="right">35</td>
<td align="right">4848</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1120_H_dinops_Gatokae.fq.gz</td>
<td align="right">10.65</td>
<td align="right">36</td>
<td align="right">3212</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1121_H_dinops_Gatokae.fq.gz</td>
<td align="right">9.18</td>
<td align="right">35</td>
<td align="right">6874</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1122_H_dinops_Gatokae.fq.gz</td>
<td align="right">43.75</td>
<td align="right">36</td>
<td align="right">288</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1154_H_demissus_Makira.fq.gz</td>
<td align="right">15.44</td>
<td align="right">36</td>
<td align="right">5201</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1167_H_diadema_Guadalcanal.fq.gz</td>
<td align="right">38.43</td>
<td align="right">37</td>
<td align="right">1379</td>
<td align="left">145</td>
</tr>
<tr class="odd">
<td align="left">THL1221_H_diadema_Gatokae.fq.gz</td>
<td align="right">39.53</td>
<td align="right">40</td>
<td align="right">43</td>
<td align="left">145</td>
</tr>
<tr class="even">
<td align="left">THL1223_H_dinops_Guadalcanal.fq.gz</td>
<td align="right">14.46</td>
<td align="right">36</td>
<td align="right">6655</td>
<td align="left">145</td>
</tr>
</tbody>
</table>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
