<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Reproducible example • RADstackshelpR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Reproducible example">
<meta property="og:description" content="RADstackshelpR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RADstackshelpR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/brown.trout.validation.html">Brown Trout Validation</a>
    </li>
    <li>
      <a href="../articles/optimize_denovo.html">Optimize denovo STACKS pipeline for Hipposideros bats</a>
    </li>
    <li>
      <a href="../articles/reproducible.vignette.html">Reproducible example</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Reproducible example</h1>
                        <h4 class="author">Devon DeRaad</h4>
            
            <h4 class="date">6 August 2021</h4>
      
      
      <div class="hidden name"><code>reproducible.vignette.Rmd</code></div>

    </div>

    
    
<div id="this-vignettearticle-is-designed-to-provide-a-fully-reproducible-example-of-how-to-use-the-radstackshelpr-package-to-optimize-stacks-parameters--each-bash-code-chunk-gives-a-generic-example-of-code-you-could-use-to-execute-the-stacks-pipeline--each-r-code-chunk-uses-extremely-light-20-samples-200-snps-toy-vcf-files-which-are-distributed-with-the-package-in-order-to-provide-fully-reproducible-examples-for-end-users--these-chunks-should-run-successfully-if-you-have-successfully-installed-the-package-" class="section level3">
<h3 class="hasAnchor">
<a href="#this-vignettearticle-is-designed-to-provide-a-fully-reproducible-example-of-how-to-use-the-radstackshelpr-package-to-optimize-stacks-parameters--each-bash-code-chunk-gives-a-generic-example-of-code-you-could-use-to-execute-the-stacks-pipeline--each-r-code-chunk-uses-extremely-light-20-samples-200-snps-toy-vcf-files-which-are-distributed-with-the-package-in-order-to-provide-fully-reproducible-examples-for-end-users--these-chunks-should-run-successfully-if-you-have-successfully-installed-the-package-" class="anchor"></a>This vignette/article is designed to provide a fully reproducible example of how to use the RADstackshelpR package to optimize <a href="https://catchenlab.life.illinois.edu/stacks/">STACKS</a> parameters. Each bash code chunk gives a generic example of code you could use to execute the STACKS pipeline. Each R code chunk uses extremely light (20 samples, ~200 SNPs), toy vcf files which are distributed with the package, in order to provide fully reproducible examples for end-users. These chunks should run successfully if you have successfully installed the package.</h3>
</div>
<div id="demultiplex" class="section level3">
<h3 class="hasAnchor">
<a href="#demultiplex" class="anchor"></a>Demultiplex</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">/home/path/to/stacks-2.41/process_radtags</span> -p .  -o . -b plate.1.barcodes.txt -e ndeI -r -c -q</code></pre></div>
<p>More details on demultiplexing using process_radtags can be found <a href="https://catchenlab.life.illinois.edu/stacks/comp/process_radtags.php">here</a></p>
</div>
<div id="iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module" class="section level3">
<h3 class="hasAnchor">
<a href="#iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module" class="anchor"></a>Iterate over potential values for the 'm' parameter in the 'ustacks' module</h3>
<p>Once you have an individual zipped fastq file for each sample, we need to iterate over the relevant values for 'm' within the 'ustacks' module (here using 15 threads at each step to speed up computation). Running the following code in a terminal window will perform five separate iterations of the entire STACKS pipeline, each with a different parameter setting for 'm' (3-7), and save the results as an unfiltered vcf file in a specified directory.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">#designate all sample ID's to a single variable called 'files', each sample should be in the directory, and the filename should match this designation except for the extension, e.g., 'sample_2' = 'sample_2.fq.gz'</span>
<span class="va">files=</span><span class="st">"sample_1</span>
<span class="st">sample_2</span>
<span class="st">sample_3</span>
<span class="st">sample_4</span>
<span class="st">sample_5"</span>
<span class="co"># Build loci de novo in each sample for the single-end reads only.</span>
<span class="co"># -M — Maximum distance (in nucleotides) allowed between stacks (default 2).</span>
<span class="co"># -m — Minimum depth of coverage required to create a stack (default 3).</span>
<span class="co">#here, we will vary m from 3-7, and leave all other paramaters default</span>
<span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{3..7}</span>
<span class="kw">do</span>
<span class="co">#create a directory to hold this unique iteration:</span>
<span class="fu">mkdir</span> stacks_m<span class="va">$i</span>
<span class="co">#run ustacks with m equal to the current iteration (3-7) for each sample</span>
<span class="va">id=</span>1
<span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span>
<span class="kw">do</span>
    <span class="ex">/home/path/to/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_m<span class="va">$i</span> -i <span class="va">$id</span> -m <span class="va">$i</span> -p 15
    <span class="bu">let</span> <span class="st">"id+=1"</span>
<span class="kw">done</span>
<span class="co">## Run cstacks to compile stacks between samples. Popmap is a file in working directory called 'pipeline_popmap.txt'</span>
<span class="ex">/home/path/to/stacks-2.41/cstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="co">## Run sstacks. Match all samples supplied in the population map against the catalog.</span>
<span class="ex">/home/path/to/stacks-2.41/sstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="co">## Run tsv2bam to transpose the data so it is stored by locus, instead of by sample.</span>
<span class="ex">/home/path/to/stacks-2.41/tsv2bam</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="co">## Run gstacks: build a paired-end contig from the metapopulation data (if paired-reads provided),</span>
<span class="co">## align reads per sample, call variant sites in the population, genotypes in each individual.</span>
<span class="ex">/home/path/to/stacks-2.41/gstacks</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="co">## Run populations completely unfiltered and output unfiltered vcf, for input to the RADstackshelpR package</span>
<span class="ex">/home/path/to/stacks-2.41/populations</span> -P stacks_m<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15
<span class="kw">done</span></code></pre></div>
<p>You should now have five directories, named: stacks_m3, stacks_m4, stacks_m5, stacks_m6, &amp; stacks_m7, each of which contains an vcf file with all called SNPs for the given parameter settings (i.e., stacks_m3 = the directory containing output from the iteration where 'm' was set to 3). Now we will use RADstackshelpR to determine which of these parameter settings (m = 3-7) is optimal for this dataset according to the 'R80' cutoff (see <a href="https://doi.org/10.1111/2041-210X.12775">Lost in Parameter Space</a>). I have now moved each vcf file into a local directory, and named it according to the parameter settings for the given run.</p>
</div>
<div id="use-radstackshelpr-to-visualize-the-output-of-these-5-runs-and-determine-the-optimal-value-for-the-parameter-m-" class="section level3">
<h3 class="hasAnchor">
<a href="#use-radstackshelpr-to-visualize-the-output-of-these-5-runs-and-determine-the-optimal-value-for-the-parameter-m-" class="anchor"></a>Use RADstackshelpR to visualize the output of these 5 runs and determine the optimal value for the parameter 'm'.</h3>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load RADstackshelpR package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RADstackshelpR</span><span class="op">)</span>
<span class="co">#optimize_m function will generate summary stats on your 5 iterative runs</span>
<span class="co">#input can be full path to each file, or just the file name if the vcf files are in your working directory</span>
<span class="va">m.out</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/optimize_m.html">optimize_m</a></span><span class="op">(</span>m3 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"m3.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     m4 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"m4.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     m5 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"m5.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     m6 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"m6.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     m7 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"m7.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#Assigning the output of this function to the variable 'm.out' should generate a list containing five objects of class 'data.frame' with the following characteristics: 'depth' showing depth per sample for each m value, 'snp' showing the number of non-missing SNPs retained in each sample at each m value, 'loci' showing the number of non-missing loci retained in each sample at each m value, 'snp.R80' showing the total number of SNPs retained at an 80% completeness cutoff, and 'loci.R80' showing the total number of polymorphic loci retained at an 80% completeness cutoff.</span>
<span class="co">#Use this output list as input for this function, to visualize the effect of varying m on the depth of each sample</span>
<span class="fu"><a href="../reference/vis_depth.html">vis_depth</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span><span class="op">)</span>
<span class="co">#&gt; [1] "Visualize how different values of m affect average depth in each sample"</span>
<span class="co">#&gt; Picking joint bandwidth of 11.9</span></code></pre></div>
<p><img src="reproducible.vignette_files/figure-html/unnamed-chunk-3-1.png" width="525"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying m on the number of SNPs retained</span>
<span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 8.81</span></code></pre></div>
<p><img src="reproducible.vignette_files/figure-html/unnamed-chunk-3-2.png" width="525"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying m on the number of loci retained</span>
<span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">m.out</span>, stacks_param <span class="op">=</span> <span class="st">"m"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of m affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 8.81</span></code></pre></div>
<p><img src="reproducible.vignette_files/figure-html/unnamed-chunk-3-3.png" width="525"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#3 is the optimal m value, and will be used next to optimize M</span></code></pre></div>
</div>
<div id="iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module-1" class="section level3">
<h3 class="hasAnchor">
<a href="#iterate-over-potential-values-for-the-m-parameter-in-the-ustacks-module-1" class="anchor"></a>Iterate over potential values for the 'M' parameter in the 'ustacks' module</h3>
<p>Now that we know the optimal value for 'm' is 3, we will repeat the process of iterating over parameter values in STACKS, this time varying the 'M' parameter from 1-8 within the 'ustacks' module. In this example, we again use 15 threads to speed up each step. Execute the following code in a terminal window:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># -M — Maximum distance (in nucleotides) allowed between stacks (default 2).</span>
<span class="co"># -m — Minimum depth of coverage required to create a stack (default 3).</span>
<span class="co">#here, vary M from 1-8, and set m to the optimized value based on prior visualizations (here 3)</span>
<span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{1..8}</span>
<span class="kw">do</span>
<span class="co">#create a directory to hold this unique iteration:</span>
<span class="fu">mkdir</span> stacks_bigM<span class="va">$i</span>
<span class="co">#run ustacks with M equal to the current iteration (1-8) for each sample, and m set to the optimized value (here, m=3)</span>
<span class="va">id=</span>1
<span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span>
<span class="kw">do</span>
    <span class="ex">/home/path/to/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_bigM<span class="va">$i</span> -i <span class="va">$id</span> -m 3 -M <span class="va">$i</span> -p 15
    <span class="bu">let</span> <span class="st">"id+=1"</span>
<span class="kw">done</span>
<span class="ex">/home/path/to/stacks-2.41/cstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/path/to/stacks-2.41/sstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/path/to/stacks-2.41/tsv2bam</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/path/to/stacks-2.41/gstacks</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/path/to/stacks-2.41/populations</span> -P stacks_bigM<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15
<span class="kw">done</span></code></pre></div>
<p>You should now have eight directories, named: stacks_bigM1, stacks_bigM2, ... stacks_bigM8,, each of which contains an vcf file with all called SNPs for the given parameter settings. I have again moved each vcf file into a local directory, and named it according to the parameter settings for the given run.</p>
</div>
<div id="use-radstackshelpr-to-visualize-the-output-of-these-8-runs-and-determine-the-optimal-value-for-the-parameter-m-" class="section level3">
<h3 class="hasAnchor">
<a href="#use-radstackshelpr-to-visualize-the-output-of-these-8-runs-and-determine-the-optimal-value-for-the-parameter-m-" class="anchor"></a>Use RADstackshelpR to visualize the output of these 8 runs and determine the optimal value for the parameter 'M'.</h3>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#optimize_bigM function will generate summary stats on your 8 iterative runs</span>
<span class="va">M.out</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/optimize_bigM.html">optimize_bigM</a></span><span class="op">(</span>M1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM1.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     M2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM2.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     M3 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM3.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     M4 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM4.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     M5 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM5.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     M6 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM6.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     M7 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM7.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                     M8 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"bigM8.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#Assigning the output of this function to the variable 'M.out' should generate a list containing four objects of class 'data.frame' with the following characteristics: 'snp' showing the number of non-missing SNPs retained in each sample at each m value, 'loci' showing the number of non-missing loci retained in each sample at each m value, 'snp.R80' showing the total number of SNPs retained at an 80% completeness cutoff, and 'loci.R80' showing the total number of polymorphic loci retained at an 80% completeness cutoff.</span>
<span class="co">#use this function to visualize the effect of varying 'M' on the number of SNPs retained</span>
<span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 7.08</span></code></pre></div>
<p><img src="reproducible.vignette_files/figure-html/unnamed-chunk-5-1.png" width="525"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying 'M' on the number of polymorphic loci retained</span>
<span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">M.out</span>, stacks_param <span class="op">=</span> <span class="st">"M"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of M affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 7.08</span></code></pre></div>
<p><img src="reproducible.vignette_files/figure-html/unnamed-chunk-5-2.png" width="525"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#optimal value for this dataset is M = 2</span></code></pre></div>
</div>
<div id="iterate-over-potential-values-for-the-n-parameter-in-the-cstacks-module" class="section level3">
<h3 class="hasAnchor">
<a href="#iterate-over-potential-values-for-the-n-parameter-in-the-cstacks-module" class="anchor"></a>Iterate over potential values for the 'n' parameter in the 'cstacks' module</h3>
<p>For the last optimization iteration, we will vary the 'n' parameter within the 'cstacks' module, using the previously optimized values for 'm' and 'M', by running the following code in a terminal window:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># -n — Number of mismatches allowed between sample loci when build the catalog (default 1).</span>
<span class="co">#here, vary 'n' across M-1, M, and M+1 (because my optimized 'M' value = 2, I will iterate over 1, 2, and 3 here), with 'm' and 'M' set to the optimized value based on prior visualizations (here 'm' = 3, and 'M'=2).</span>
<span class="kw">for</span> <span class="ex">i</span> in <span class="dt">{1..3}</span>
<span class="kw">do</span>
<span class="co">#create a directory to hold this unique iteration:</span>
<span class="fu">mkdir</span> stacks_n<span class="va">$i</span>
<span class="co">#run ustacks with n equal to the current iteration (1-3) for each sample, m = 3, and M=2</span>
<span class="va">id=</span>1
<span class="kw">for</span> <span class="ex">sample</span> in <span class="va">$files</span>
<span class="kw">do</span>
    <span class="ex">/home/path/to/stacks-2.41/ustacks</span> -f <span class="va">${sample}</span>.fq.gz -o stacks_n<span class="va">$i</span> -i <span class="va">$id</span> -m 3 -M 2 -p 15
    <span class="bu">let</span> <span class="st">"id+=1"</span>
<span class="kw">done</span>
<span class="ex">/home/path/to/stacks-2.41/cstacks</span> -n <span class="va">$i</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/path/to/stacks-2.41/sstacks</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -p 15
<span class="ex">/home/path/to/stacks-2.41/tsv2bam</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/path/to/stacks-2.41/gstacks</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt -t 15
<span class="ex">/home/path/to/stacks-2.41/populations</span> -P stacks_n<span class="va">$i</span> -M pipeline_popmap.txt --vcf -t 15
<span class="kw">done</span></code></pre></div>
</div>
<div id="use-radstackshelpr-to-visualize-the-output-of-these-3-runs-and-determine-the-optimal-value-for-n-" class="section level3">
<h3 class="hasAnchor">
<a href="#use-radstackshelpr-to-visualize-the-output-of-these-3-runs-and-determine-the-optimal-value-for-n-" class="anchor"></a>Use RADstackshelpR to visualize the output of these 3 runs and determine the optimal value for 'n'.</h3>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#optimize n</span>
<span class="va">n.out</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/optimize_n.html">optimize_n</a></span><span class="op">(</span>nequalsMminus1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"nequalsmminus1.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                        nequalsM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"nequalsm.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span>,
                        nequalsMplus1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"nequalsmplus1.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"RADstackshelpR"</span><span class="op">)</span><span class="op">)</span>
<span class="co">##Assigning the output of this function to the variable 'n.out' should generate a single object of class 'data.frame' showing the number of SNPs and loci retained across filtering levels for each value of n.</span>
<span class="co">#visualize the effect of varying n on the number of SNPs retained</span>
<span class="fu"><a href="../reference/vis_snps.html">vis_snps</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of SNPs retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of SNPs retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of SNPs retained at an 80% completeness cutoff.</span>
<span class="co">#&gt; Picking joint bandwidth of 4.71</span></code></pre></div>
<p><img src="reproducible.vignette_files/figure-html/unnamed-chunk-7-1.png" width="525"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize the effect of varying n on the number of polymorphic loci retained</span>
<span class="fu"><a href="../reference/vis_loci.html">vis_loci</a></span><span class="op">(</span>output <span class="op">=</span> <span class="va">n.out</span>, stacks_param <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span>
<span class="co">#&gt; Visualize how different values of n affect number of polymorphic loci retained.</span>
<span class="co">#&gt; Density plot shows the distribution of the number of loci retained in each sample,</span>
<span class="co">#&gt; while the asterisk denotes the total number of loci retained at an 80% completeness cutoff. The optimal value is denoted by red color.</span>
<span class="co">#&gt; Picking joint bandwidth of 4.71</span></code></pre></div>
<p><img src="reproducible.vignette_files/figure-html/unnamed-chunk-7-2.png" width="525"></p>
<p>You now have a de novo-assembled, parameter-optimized, unfiltered SNP dataset for your organism, and a single cohesive figure documenting the filtering process, which you can stick in your paper's supplement or host on your own repository. The next step is to filter this vcf file using the SNPfiltR package, which is specifically designed to pick up where this pipeline leaves off, or with any other variant filtering program of your choice.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
